L1Trigger/TrackFindingTracklet/interface/CircularBuffer.h

```diff
@@ -47,8 +52,13 @@ namespace trklet {
       assert(!full());
       buffer_[wptr_++] = element;
       wptr_ = wptr_ % size_;
+      assert(wptr_ != rptr_);
     }
 
+    unsigned int rptr() const { return rptr_; }
+
+    unsigned int wptr() const { return wptr_; }
+
```

Comment:
Where are rptr() and wptr() needed?

Reply:
These methods are very handy when debugging the code and comparing to the HLS code to make sure you have agreement.

Reply:
added comment to clarify